src/lib/mina/errors.ts

```diff
@@ -60,23 +60,26 @@ function invalidTransactionError(
 ): string {
   let errorMessages = [];
   let rawErrors = JSON.stringify(errors);
+  let n = transaction.accountUpdates.length;
+  let accountUpdateErrors = errors.slice(1, n + 1);
 
-  // handle errors for fee payer
-  let errorsForFeePayer = errors[0];
-  for (let [error] of errorsForFeePayer) {
-    let message = ErrorHandlers[error as keyof typeof ErrorHandlers]?.({
-      transaction,
-      accountUpdateIndex: NaN,
-      isFeePayer: true,
-      ...additionalContext,
-    });
-    if (message) errorMessages.push(message);
+  // Check if the number of errors match the number of account updates. If there are more, then the fee payer has an error.
+  // We do this check because the fee payer error is not included in network transaction errors and is always present (even if empty) in the local transaction errors.
+  if (accountUpdateErrors.length === n) {
+    let errorsForFeePayer = errors[0];
```

Comment:
after staring at this long enough the logic looks correct, because you slice off the account update errors before and the slice starting from 1 can be either `n` or `n-1` entries long.

but would be more readable to do
```ts
if (errors.length > n) {
   let errorsForFeePayer = errors.unshift();
   // ...
```

and then you can reuse the `errors` below!

Reply:
great idea! fixed!